### ISSUE TITLE
[Bug]: Pin buffers are not upgraded anymore (local.manual_groupings)

### ISSUE THREAD

**Issue by otavioschwanck** [akinsho/bufferline.nvim#691]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

If i try to unpin some buffer, the g:BufferlinePinnedBuffers keeps the same.  If i restart the session, it continues on pinned.

### What did you expect to happen?

remove completely from pinned

### Config

```
return {
  {
    -- "akinsho/bufferline.nvim",
    dir = "~/Projetos/bufferline.nvim/",
    lazy = false,
    keys = {
      { "<leader>bp", "<Cmd>BufferLineTogglePin<CR>",                                                desc = "Toggle pin" },
      { "<leader>bk", "<Cmd>BufferLineGroupClose ungrouped<CR>",                                     desc = "Delete non-pinned buffers" },
      { "<leader>bK", "<Cmd>BufferLineGroupClose ungrouped<CR><Cmd>BufferLineGroupClose pinned<CR>", desc = "Delete all buffers" },
      { "<leader>bn", ":e ~/.nvim-scratch<CR>",                                                      desc = "Open Scratch Buffer" },
      { "<C-s>",      "<cmd>BufferLineTogglePin<cr>",                                                desc = "Prev Buffer" },
      { "[b",         "<cmd>BufferLineCyclePrev<cr>",                                                desc = "Prev Buffer" },
      { "]b",         "<cmd>BufferLineCycleNext<cr>",                                                desc = "Next Buffer" },
      { "H",          "<cmd>BufferLineCyclePrev<cr>",                                                desc = "Prev Buffer" },
      { "L",          "<cmd>BufferLineCycleNext<cr>",                                                desc = "Next Buffer" },
      { "[B",         "<cmd>BufferLineMovePrev<cr>",                                                 desc = "Switch Prev Buffer" },
      { "]B",         "<cmd>BufferLineMoveNext<cr>",                                                 desc = "Switch Next Buffer" },
      { "<C-h>",      "<cmd>BufferLineMovePrev<cr>",                                                 desc = "Switch Prev Buffer" },
      { "<C-l>",      "<cmd>BufferLineMoveNext<cr>",                                                 desc = "Switch Next Buffer" },
    },
    opts = {
      options = {
        diagnostics = "nvim_lsp",
        always_show_bufferline = true,
        offsets = {
          {
            filetype = "neo-tree",
            text = "Neo-tree",
            highlight = "Directory",
            text_align = "left",
          },
        },
      },
    },
  }
}
```

### Additional Information

![bug](https://user-images.githubusercontent.com/9551316/220366909-e35b645c-ca6f-478a-be04-65382b2332a8.gif)

### commit

cbb798dd2db7841550cd2c6c6dde12dfda055928

**Comment by akinsho** [comment]
@otavioschwanck can you please make sure you are running the latest `dev` commit, and retest this. You're minimal config is pointing to a local build which is not valid for raising issues

**Comment by akinsho** [comment]
@otavioschwanck I'm finding all these issues you are opening very hard to follow. You are opening quite a lot of them without very well explain mechanisms to reproduce these issues and only partial explanations on how to reproduce them. Please try to spend a bit more time given an itemised list of how to reproduce a bug e.g. 1.) open neovim with -u minimal/config 2.)...

secondly if issues are related please just create one rather than 10 there aren't 10 maintainers just me so more issues is not better for me just harder to track

**Comment by akinsho** [comment]
In the case of this issue can you please go back to before this bug started happening and find out which commit introduced this specifically this is what the commit field in the issue form is for i.e. which commit introduced the bug? is it related to the other contribution you made? was it working before that.

**Comment by otavioschwanck** [comment]
The commit the make this error happen:

6dbb3d44a6149b1919f7b13e117416fb18f0814b

**Comment by akinsho** [comment]
fixed by #693
